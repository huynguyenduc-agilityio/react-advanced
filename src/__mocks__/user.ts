// Icons
import { KebabIcon, HeartIcon, UserGroupIcon, UserIcon } from '@/components';

// Types

import { IUserModel, NotificationType, Status as StatusEnum } from '@/types';

export const mockInitialValuesUser: IUserModel = {
  name: 'John Doe',
  email: 'john.doe@example.com',
  phone: '123456789',
  position: 'Software Engineer',
  location: 'New York',
  website: 'https://johndoe.com',
  company: 'Tech Corp',
  description: 'Passionate about coding.',
  teamName: 'Google',
  rank: 'marketing',
  office: 'New York, NY',
  teamMail: 'dashdark@gmail.com',
  payment: 'VISA',
  billName: 'John Carter',
  billAddress: '601 4th St #103',
  state: 'Los Angeles',
  zipCode: '90001',
  mentionMessage: NotificationType.InApp,
  replyMessage: NotificationType.Email,
  assignTask: NotificationType.Email,
  taskOverdue: NotificationType.InApp,
  dailySummary: NotificationType.Email,
  weeklySummary: NotificationType.InApp,
  monthlySummary: NotificationType.InApp,
  annuallySummary: NotificationType.Email,
  status: StatusEnum.Online,
};

export const mockUsers = {
  items: [
    {
      id: '1',
      name: 'John Doe',
      avatar: '',
      email: 'john@example.com',
      phone: '123-456-7890',
      location: 'New York',
      company: 'Tech Inc',
      status: 'active',
    },
  ],
  meta: { pagination: { total: 1, pageCount: 1 } },
};

export const MOCK_SUMMARY_DASHBOARD = [
  {
    id: '1',
    title: 'Total Users',
    total: 250,
    icon: UserGroupIcon,
  },
  {
    id: '2',
    title: 'New Users',
    total: 15,
    icon: UserIcon,
  },
  {
    id: '3',
    title: 'Top Users',
    total: 200,
    icon: HeartIcon,
  },
  {
    id: '4',
    title: 'Other Users',
    total: 35,
    icon: KebabIcon,
  },
];
