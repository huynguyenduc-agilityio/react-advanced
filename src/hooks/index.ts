export * from './usePagination';
export * from './useUpload';
export * from './useUser';
export * from './useCustomToast';
