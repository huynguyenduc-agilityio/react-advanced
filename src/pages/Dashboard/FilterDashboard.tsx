import { memo, useCallback, useDeferredValue } from 'react';
import isEqual from 'react-fast-compare';
import { Link } from 'react-router-dom';
import { Button, Flex, Text } from '@chakra-ui/react';

import { DEFAULT_CURRENT_PAGE, PUBLIC_ROUTERS } from '@/constants';

import { fonts } from '@/themes/bases/typography';

import { SearchBox } from '@/components';
import { useUserFilterStore } from '@/stores';

const FilterDashboard = ({
  onSetPage,
}: {
  onSetPage: (page: number) => void;
}) => {
  const { search, setSearch } = useUserFilterStore();
  const deferredQuery = useDeferredValue(search);

  const handleSearch = useCallback(
    (value: string) => {
      setSearch(value);
      onSetPage(DEFAULT_CURRENT_PAGE);
    },
    [setSearch],
  );
  return (
    <Flex justifyContent="space-between" width="full">
      <Flex alignItems="center" gap="50px">
        <Text
          fontFamily={fonts.heading}
          fontSize="xl"
          fontWeight="bold"
          color="white"
        >
          Users
        </Text>

        <SearchBox
          defaultValue={deferredQuery}
          onChange={handleSearch}
          w="352px"
        />
      </Flex>

      <Link to={PUBLIC_ROUTERS.USER_ADD}>
        <Button variant="primary" title="Add User" width="137px">
          Add User
        </Button>
      </Link>
    </Flex>
  );
};

export default memo(FilterDashboard, isEqual);
