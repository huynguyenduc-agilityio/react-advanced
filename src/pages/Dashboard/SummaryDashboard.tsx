import { memo } from 'react';
import { Flex } from '@chakra-ui/react';

// Mocks
import { MOCK_SUMMARY_DASHBOARD } from '@/__mocks__';

// Components
import { CardSummary } from '@/components';

const SummaryDashboard = () => {
  return (
    <Flex w="full" gap={8} mt="50px" mb="30px">
      {MOCK_SUMMARY_DASHBOARD.map(({ title, total, icon: Icon }) => (
        <CardSummary key={title} title={title} total={total} icon={<Icon />} />
      ))}
    </Flex>
  );
};

export default memo(SummaryDashboard);
