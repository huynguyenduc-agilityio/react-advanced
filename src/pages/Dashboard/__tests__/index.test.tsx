import { MemoryRouter, useSearchParams } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { render, screen, waitFor } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { ChakraProvider } from '@chakra-ui/react';

// Hooks
import { useGetUsers } from '@/hooks';

// Mocks
import { mockUsers } from '@/__mocks__';

// Pages
import Dashboard from '..';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: jest.fn().mockReturnValue({ pathname: '/dashboard' }),
  useSearchParams: jest.fn(() => [new URLSearchParams(), jest.fn()]),
}));

jest.mock('@/hooks', () => ({
  useGetUsers: jest.fn(),
  useDeleteUser: jest.fn(() => ({
    handleDeleteUser: jest.fn(),
    isDeleteLoading: false,
  })),
  useDeleteMultipleUsers: jest.fn(() => ({
    handleDeleteMultipleUsers: jest.fn(),
    isDeleteMultipleLoading: false,
  })),
  usePagination: jest.fn(() => ({
    isDisableNext: false,
    isDisablePrev: false,
    handleChangeLimit: jest.fn(),
    handleChangePage: jest.fn(),
  })),
  useCustomToast: jest.fn(() => jest.fn()),
}));

describe('Dashboard Component', () => {
  const queryClient = new QueryClient();
  let mockSetSearchParams: jest.Mock;

  beforeEach(() => {
    mockSetSearchParams = jest.fn();
    (useSearchParams as jest.Mock).mockReturnValue([
      new URLSearchParams(),
      mockSetSearchParams,
    ]);

    (useGetUsers as jest.Mock).mockReturnValue({
      users: mockUsers,
      isUsersLoading: false,
    });
  });

  const renderComponent = () =>
    render(
      <QueryClientProvider client={queryClient}>
        <ChakraProvider>
          <MemoryRouter initialEntries={['/dashboard']}>
            <Dashboard />
          </MemoryRouter>
        </ChakraProvider>
      </QueryClientProvider>,
    );

  it('should match snapshot', () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });

  it('renders Dashboard correctly', () => {
    renderComponent();
    expect(screen.getByText('Users')).toBeInTheDocument();
    expect(screen.getByText('John Doe')).toBeInTheDocument();
  });

  it('updates search query when input has value', async () => {
    renderComponent();
    const searchBox = screen.getByRole('textbox');

    await userEvent.type(searchBox, 'Jane');
    await waitFor(() =>
      expect(mockSetSearchParams).toHaveBeenCalledWith(
        expect.any(URLSearchParams),
      ),
    );
  });

  it('clears search query when input is empty', async () => {
    renderComponent();
    const searchBox = screen.getByRole('textbox');

    await userEvent.type(searchBox, 'Jane');
    await waitFor(() => expect(mockSetSearchParams).toHaveBeenCalled());

    await userEvent.clear(searchBox);
    await waitFor(() => {
      expect(mockSetSearchParams).toHaveBeenCalledWith(new URLSearchParams());
    });
  });

  it('navigates to add user page', () => {
    renderComponent();
    const addUserButton = screen.getByText('Add User');
    expect(addUserButton.closest('a')).toHaveAttribute(
      'href',
      '/users/add-user',
    );
  });
});
