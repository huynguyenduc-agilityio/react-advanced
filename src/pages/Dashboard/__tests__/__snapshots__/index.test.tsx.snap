// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Component should render successfully 1`] = `
<div>
  <div
    class="chakra-stack css-mw8ufx"
  >
    <div
      class="css-e03czl"
    >
      <div
        class="css-br9knx"
      >
        <p
          class="chakra-text css-q54xsm"
        >
          Users
        </p>
        <div
          class="chakra-input__group css-165svw3"
          data-group="true"
        >
          <input
            class="chakra-input css-1vweg4"
            placeholder="Search for..."
            type="text"
            value=""
          />
          <div
            class="chakra-input__left-element css-t7swz3"
          >
            <svg
              class="chakra-icon css-gxpt7g"
              fill="currentColor"
              focusable="false"
              height="1em"
              id="search"
              stroke="currentColor"
              stroke-width="0"
              version="1.1"
              viewBox="0 0 24 24"
              width="1em"
              x="0px"
              xmlns="http://www.w3.org/2000/svg"
              y="0px"
            >
              <g>
                <path
                  d="M20.031,20.79c0.46,0.46,1.17-0.25,0.71-0.7l-3.75-3.76c1.27-1.41,2.04-3.27,2.04-5.31
		c0-4.39-3.57-7.96-7.96-7.96s-7.96,3.57-7.96,7.96c0,4.39,3.57,7.96,7.96,7.96c1.98,0,3.81-0.73,5.21-1.94L20.031,20.79z
		 M4.11,11.02c0-3.84,3.13-6.96,6.96-6.96c3.84,0,6.96,3.12,6.96,6.96c0,3.84-3.12,6.96-6.96,6.96C7.24,17.98,4.11,14.86,4.11,11.02
		z"
                />
              </g>
            </svg>
          </div>
        </div>
      </div>
      <a
        href="/users/add-user"
      >
        <button
          class="chakra-button css-14e96w5"
          title="Add User"
          type="button"
        >
          Add User
        </button>
      </a>
    </div>
    <div
      class="css-cjropt"
    >
      <div
        class="css-1ir6bz1"
      >
        <div
          class="css-1bejyq4"
        >
          <div
            class="css-0"
          >
            <svg
              fill="none"
              height="40"
              viewBox="0 0 41 40"
              width="41"
            >
              <rect
                fill="#CB3CFF"
                height="40"
                opacity="0.2"
                rx="19.908"
                width="39.8159"
                x="0.263672"
              />
              <path
                d="M17.3359 19.4798C17.8457 19.4798 18.313 19.243 18.691 18.8792C18.2945 18.1682 18.0682 17.3638 18.0682 16.592C18.0682 15.9374 18.2261 15.32 18.504 14.7748C18.1593 14.5634 17.7564 14.4395 17.3242 14.4395C16.061 14.4395 15.0371 15.4783 15.0371 16.7595C15.0371 18.0408 16.0727 19.4798 17.3359 19.4798Z"
                fill="#CB3CFF"
              />
              <path
                d="M18.8742 20.58C18.6509 20.5318 18.4199 20.5052 18.1824 20.5052H16.4815C14.6715 20.5052 13.2041 21.9772 13.2041 23.7931L13.2118 24.1819C13.2118 24.6892 13.6218 25.1006 14.1276 25.1006H15.8047C15.887 23.0803 17.1252 21.3579 18.8742 20.58Z"
                fill="#CB3CFF"
              />
              <path
                d="M22.0165 19.9647C23.5827 19.9647 24.8381 18.1804 24.8381 16.5917C24.8381 15.003 23.5683 13.715 22.0021 13.715C20.4358 13.715 19.166 15.003 19.166 16.5917C19.1662 18.1804 20.4503 19.9647 22.0165 19.9647Z"
                fill="#CB3CFF"
              />
              <path
                d="M27.13 25.3137C27.13 23.0619 25.3105 21.2366 23.0662 21.2366H20.9571C18.7128 21.2366 16.8933 23.0619 16.8933 25.3137L16.903 25.7958C16.903 26.4248 17.4113 26.9349 18.0385 26.9349H26.0043C26.6314 26.9349 27.1398 26.4248 27.1398 25.7958L27.13 25.3137Z"
                fill="#CB3CFF"
              />
            </svg>
          </div>
          <div
            class="css-1q54ih0"
          >
            <div
              class="css-0"
            >
              <p
                class="chakra-text css-60ffrt"
              >
                Total Users
              </p>
              <p
                class="chakra-text css-qlr254"
              >
                250
              </p>
            </div>
            <button
              aria-label="More options"
              class="chakra-button css-q9fwed"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 6a2 2 0 110-4 2 2 0 010 4zM10 12a2 2 0 110-4 2 2 0 010 4zM10 18a2 2 0 110-4 2 2 0 010 4z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="css-1ir6bz1"
      >
        <div
          class="css-1bejyq4"
        >
          <div
            class="css-0"
          >
            <svg
              fill="none"
              height="40"
              viewBox="0 0 41 40"
              width="41"
            >
              <rect
                fill="#FDB52A"
                height="40"
                opacity="0.2"
                rx="19.908"
                width="39.8159"
                x="0.263672"
              />
              <path
                d="M13.282 25.9874C13.282 23.5366 15.2688 21.5498 17.7196 21.5498H22.624C25.0748 21.5498 27.0615 23.5366 27.0615 25.9874C27.0615 26.8043 26.3993 27.4666 25.5823 27.4666H14.7612C13.9442 27.4666 13.282 26.8043 13.282 25.9874Z"
                fill="#FDB52A"
              />
              <path
                d="M20.1744 20.4225C22.3429 20.4225 24.1008 18.6565 24.1008 16.478C24.1008 14.2995 22.3429 12.5334 20.1744 12.5334C18.0059 12.5334 16.248 14.2995 16.248 16.478C16.248 18.6565 18.0059 20.4225 20.1744 20.4225Z"
                fill="#FDB52A"
              />
            </svg>
          </div>
          <div
            class="css-1q54ih0"
          >
            <div
              class="css-0"
            >
              <p
                class="chakra-text css-60ffrt"
              >
                New Users
              </p>
              <p
                class="chakra-text css-qlr254"
              >
                15
              </p>
            </div>
            <button
              aria-label="More options"
              class="chakra-button css-q9fwed"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 6a2 2 0 110-4 2 2 0 010 4zM10 12a2 2 0 110-4 2 2 0 010 4zM10 18a2 2 0 110-4 2 2 0 010 4z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="css-1ir6bz1"
      >
        <div
          class="css-1bejyq4"
        >
          <div
            class="css-0"
          >
            <svg
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
            >
              <rect
                fill="#05C168"
                height="40"
                opacity="0.2"
                rx="19.908"
                width="39.8159"
                x="0.180176"
              />
              <path
                d="M23.6274 12.8887C22.9406 12.8887 22.2632 13.0493 21.6489 13.3579C21.0346 13.6665 20.5003 14.1145 20.0882 14.6664C19.5311 13.9202 18.7544 13.369 17.8681 13.0908C16.9818 12.8127 16.0309 12.8218 15.15 13.1167C14.2691 13.4117 13.503 13.9777 12.9601 14.7344C12.4171 15.4911 12.125 16.4003 12.125 17.3331C12.125 22.659 19.4909 26.859 19.8006 27.0368C19.8884 27.0864 19.9875 27.1119 20.0882 27.1109C20.1889 27.1126 20.2883 27.087 20.3757 27.0368C21.7322 26.2402 23.0045 25.307 24.173 24.2516C26.7463 21.9257 28.0514 19.5998 28.0514 17.3331C28.0514 16.1544 27.5853 15.0239 26.7556 14.1904C25.926 13.3569 24.8007 12.8887 23.6274 12.8887Z"
                fill="#05C168"
              />
            </svg>
          </div>
          <div
            class="css-1q54ih0"
          >
            <div
              class="css-0"
            >
              <p
                class="chakra-text css-60ffrt"
              >
                Top Users
              </p>
              <p
                class="chakra-text css-qlr254"
              >
                200
              </p>
            </div>
            <button
              aria-label="More options"
              class="chakra-button css-q9fwed"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 6a2 2 0 110-4 2 2 0 010 4zM10 12a2 2 0 110-4 2 2 0 010 4zM10 18a2 2 0 110-4 2 2 0 010 4z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="css-1ir6bz1"
      >
        <div
          class="css-1bejyq4"
        >
          <div
            class="css-0"
          >
            <svg
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
            >
              <rect
                fill="#086CD9"
                height="40"
                opacity="0.2"
                rx="19.908"
                width="39.8159"
                x="0.180176"
              />
              <path
                d="M20.0882 12C18.5132 12 16.9736 12.4692 15.6641 13.3482C14.3545 14.2273 13.3339 15.4767 12.7312 16.9385C12.1285 18.4003 11.9708 20.0089 12.278 21.5607C12.5853 23.1126 13.3437 24.538 14.4574 25.6569C15.571 26.7757 16.9899 27.5376 18.5346 27.8463C20.0794 28.155 21.6805 27.9965 23.1356 27.391C24.5906 26.7855 25.8343 25.7602 26.7093 24.4446C27.5843 23.129 28.0514 21.5822 28.0514 20C28.0473 17.8795 27.207 15.847 25.7145 14.3476C24.222 12.8482 22.1989 12.0041 20.0882 12ZM16.4129 20.9231C16.2311 20.9231 16.0535 20.8689 15.9024 20.7675C15.7513 20.6661 15.6335 20.5219 15.564 20.3532C15.4944 20.1846 15.4762 19.999 15.5117 19.8199C15.5472 19.6409 15.6347 19.4764 15.7632 19.3473C15.8917 19.2182 16.0554 19.1303 16.2336 19.0947C16.4119 19.059 16.5966 19.0773 16.7645 19.1472C16.9324 19.2171 17.0759 19.3354 17.1768 19.4872C17.2778 19.639 17.3317 19.8174 17.3317 20C17.3317 20.2448 17.2349 20.4796 17.0626 20.6527C16.8903 20.8258 16.6566 20.9231 16.4129 20.9231ZM20.0882 20.9231C19.9065 20.9231 19.7288 20.8689 19.5777 20.7675C19.4266 20.6661 19.3088 20.5219 19.2393 20.3532C19.1698 20.1846 19.1516 19.999 19.187 19.8199C19.2225 19.6409 19.31 19.4764 19.4385 19.3473C19.567 19.2182 19.7307 19.1303 19.9089 19.0947C20.0872 19.059 20.2719 19.0773 20.4398 19.1472C20.6077 19.2171 20.7512 19.3354 20.8522 19.4872C20.9531 19.639 21.007 19.8174 21.007 20C21.007 20.2448 20.9102 20.4796 20.7379 20.6527C20.5656 20.8258 20.3319 20.9231 20.0882 20.9231ZM23.7635 20.9231C23.5818 20.9231 23.4041 20.8689 23.253 20.7675C23.1019 20.6661 22.9842 20.5219 22.9146 20.3532C22.8451 20.1846 22.8269 19.999 22.8623 19.8199C22.8978 19.6409 22.9853 19.4764 23.1138 19.3473C23.2423 19.2182 23.406 19.1303 23.5842 19.0947C23.7625 19.059 23.9472 19.0773 24.1151 19.1472C24.283 19.2171 24.4265 19.3354 24.5275 19.4872C24.6284 19.639 24.6823 19.8174 24.6823 20C24.6823 20.2448 24.5855 20.4796 24.4132 20.6527C24.2409 20.8258 24.0072 20.9231 23.7635 20.9231Z"
                fill="#086CD9"
              />
            </svg>
          </div>
          <div
            class="css-1q54ih0"
          >
            <div
              class="css-0"
            >
              <p
                class="chakra-text css-60ffrt"
              >
                Other Users
              </p>
              <p
                class="chakra-text css-qlr254"
              >
                35
              </p>
            </div>
            <button
              aria-label="More options"
              class="chakra-button css-q9fwed"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 6a2 2 0 110-4 2 2 0 010 4zM10 12a2 2 0 110-4 2 2 0 010 4zM10 18a2 2 0 110-4 2 2 0 010 4z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="chakra-table__container css-1yeu4j9"
    >
      <div
        class="css-8h9hu1"
      >
        <header
          class="chakra-text css-m1oz5p"
        >
          All Users
        </header>
      </div>
      <table
        class="chakra-table css-5605sr"
      >
        <thead
          class="css-0"
        >
          <tr
            class="css-0"
          >
            <th
              class="css-3n66do"
            >
              <div
                class="css-k008qs"
              >
                <label
                  class="chakra-checkbox css-1577qb8"
                >
                  <input
                    aria-label="Checkbox Indeterminate"
                    class="chakra-checkbox__input"
                    style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                    type="checkbox"
                    value=""
                  />
                  <span
                    aria-hidden="true"
                    class="chakra-checkbox__control css-1dnp747"
                  />
                </label>
              </div>
            </th>
            <th
              class="css-hdc1xj"
            >
              <div
                class="css-wd8hou"
              >
                <svg
                  class="chakra-icon css-kpy8va"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                  />
                </svg>
                <p
                  class="chakra-text css-13c2rt7"
                  title="Name"
                >
                  Name
                </p>
              </div>
            </th>
            <th
              class="css-hdc1xj"
            >
              <div
                class="css-wd8hou"
              >
                <svg
                  class="chakra-icon css-kpy8va"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M497.39 361.8l-112-48a24 24 0 0 0-28 6.9l-49.6 60.6A370.66 370.66 0 0 1 130.6 204.11l60.6-49.6a23.94 23.94 0 0 0 6.9-28l-48-112A24.16 24.16 0 0 0 122.6.61l-104 24A24 24 0 0 0 0 48c0 256.5 207.9 464 464 464a24 24 0 0 0 23.4-18.6l24-104a24.29 24.29 0 0 0-14.01-27.6z"
                  />
                </svg>
                <p
                  class="chakra-text css-13c2rt7"
                  title="Phone"
                >
                  Phone
                </p>
              </div>
            </th>
            <th
              class="css-hdc1xj"
            >
              <div
                class="css-wd8hou"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-kpy8va"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="m11.54 22.351.07.04.028.016a.76.76 0 0 0 .723 0l.028-.015.071-.041a16.975 16.975 0 0 0 1.144-.742 19.58 19.58 0 0 0 2.683-2.282c1.944-1.99 3.963-4.98 3.963-8.827a8.25 8.25 0 0 0-16.5 0c0 3.846 2.02 6.837 3.963 8.827a19.58 19.58 0 0 0 2.682 2.282 16.975 16.975 0 0 0 1.145.742ZM12 13.5a3 3 0 1 0 0-6 3 3 0 0 0 0 6Z"
                    fill-rule="evenodd"
                  />
                </svg>
                <p
                  class="chakra-text css-13c2rt7"
                  title="Location"
                >
                  Location
                </p>
              </div>
            </th>
            <th
              class="css-hdc1xj"
            >
              <div
                class="css-wd8hou"
              >
                <svg
                  class="chakra-icon css-kpy8va"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M372 160v-12A116.13 116.13 0 0 0 258.89 32h-5.78A116.13 116.13 0 0 0 140 148v12H52a4 4 0 0 0-4 4v300a16 16 0 0 0 16 16h384a16 16 0 0 0 16-16V164a4 4 0 0 0-4-4zm-40 0H180v-11c0-41.84 33.41-76.56 75.25-77A76.08 76.08 0 0 1 332 148z"
                  />
                </svg>
                <p
                  class="chakra-text css-13c2rt7"
                  title="Company"
                >
                  Company
                </p>
              </div>
            </th>
            <th
              class="css-hdc1xj"
            >
              <div
                class="css-wd8hou"
              >
                <svg
                  class="chakra-icon css-kpy8va"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M208,32H48A16,16,0,0,0,32,48V208a16,16,0,0,0,16,16H208a16,16,0,0,0,16-16V48A16,16,0,0,0,208,32Zm-34.34,77.66-56,56a8,8,0,0,1-11.32,0l-24-24a8,8,0,0,1,11.32-11.32L112,148.69l50.34-50.35a8,8,0,0,1,11.32,11.32Z"
                  />
                </svg>
                <p
                  class="chakra-text css-13c2rt7"
                  title="Status"
                >
                  Status
                </p>
              </div>
            </th>
            <th
              class="css-hdc1xj"
            >
              <div
                class="css-wd8hou"
              >
                <p
                  class="chakra-text css-13c2rt7"
                />
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          class="css-tdnrhj"
        >
          <tr
            class="css-0"
          >
            <td
              class="css-iutff6"
            >
              <div
                class="css-k008qs"
              >
                <label
                  class="chakra-checkbox css-1577qb8"
                >
                  <input
                    aria-label="Checkbox 1"
                    class="chakra-checkbox__input"
                    style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                    type="checkbox"
                    value=""
                  />
                  <span
                    aria-hidden="true"
                    class="chakra-checkbox__control css-1dnp747"
                  />
                </label>
              </div>
            </td>
            <td
              class="css-lgr0fh"
            >
              <div
                class="css-1q0fhs3"
              >
                <span
                  class="chakra-avatar css-1pbrvyx"
                >
                  <div
                    aria-label="John Doe"
                    class="chakra-avatar__initials css-1ebyn6"
                    role="img"
                  >
                    JD
                  </div>
                </span>
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-33m1vg"
                  >
                    John Doe
                  </p>
                  <p
                    class="chakra-text css-vuqelp"
                  >
                    john@example.com
                  </p>
                </div>
              </div>
            </td>
            <td
              class="css-lgr0fh"
            >
              123-456-7890
            </td>
            <td
              class="css-lgr0fh"
            >
              New York
            </td>
            <td
              class="css-lgr0fh"
            >
              Tech Inc
            </td>
            <td
              class="css-lgr0fh"
            >
              <div
                class="css-5enp4c"
              >
                <svg
                  fill="none"
                  height="4"
                  viewBox="0 0 4 4"
                  width="4"
                >
                  <circle
                    cx="2.4375"
                    cy="1.59375"
                    fill="#AEB9E1"
                    r="1.5"
                  />
                </svg>
                <p
                  class="chakra-text css-1iq8whn"
                >
                  Offline
                </p>
              </div>
            </td>
            <td
              class="css-lgr0fh"
            >
              <div
                class="css-1k9efnl"
              >
                <a
                  href="/users/edit/1"
                >
                  <button
                    aria-label="Edit"
                    class="chakra-button css-o142tn"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 20 20"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="m2.695 14.762-1.262 3.155a.5.5 0 0 0 .65.65l3.155-1.262a4 4 0 0 0 1.343-.886L17.5 5.501a2.121 2.121 0 0 0-3-3L3.58 13.419a4 4 0 0 0-.885 1.343Z"
                      />
                    </svg>
                  </button>
                </a>
                <button
                  aria-label="delete"
                  class="chakra-button css-o142tn"
                  data-email="john@example.com"
                  data-id="1"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7 6V3C7 2.44772 7.44772 2 8 2H16C16.5523 2 17 2.44772 17 3V6H22V8H20V21C20 21.5523 19.5523 22 19 22H5C4.44772 22 4 21.5523 4 21V8H2V6H7ZM9 4V6H15V4H9Z"
                    />
                  </svg>
                </button>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="css-7v0npi"
    >
      <div
        class="css-70qvj9"
      >
        <p
          class="chakra-text css-1im6emh"
        >
          1 - 1 of 1
        </p>
      </div>
      <div
        class="css-1guqm6j"
      >
        <div
          class="css-193m7l6"
        >
          <p
            class="chakra-text css-15qsaxk"
          >
            Rows per page:
          </p>
          <div
            class="chakra-select__wrapper css-2a4yzx"
          >
            <select
              aria-label="rows-per-page"
              class="chakra-select css-1d7l0u9"
              title="select-limit"
            >
              <option
                selected=""
                value="10"
              >
                10
              </option>
              <option
                value="25"
              >
                25
              </option>
              <option
                value="50"
              >
                50
              </option>
              <option
                value="100"
              >
                100
              </option>
            </select>
            <div
              class="chakra-select__icon-wrapper css-ptmlnu"
            >
              <svg
                aria-hidden="true"
                class="chakra-select__icon"
                focusable="false"
                role="presentation"
                style="width: 1em; height: 1em; color: currentColor;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="css-1fxprry"
        >
          <button
            class="chakra-button css-zbgeqs"
            title="prev-button"
            type="button"
          >
            <svg
              class="chakra-icon css-1pidp76"
              fill="none"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="19"
                x2="5"
                y1="12"
                y2="12"
              />
              <polyline
                points="12 19 5 12 12 5"
              />
            </svg>
          </button>
          <button
            class="chakra-button css-zbgeqs"
            title="next-button"
            type="button"
          >
            <svg
              class="chakra-icon css-1pidp76"
              fill="none"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="5"
                x2="19"
                y1="12"
                y2="12"
              />
              <polyline
                points="12 5 19 12 12 19"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
